lib/rules.js

```diff
@@ -197,9 +197,11 @@ module.exports = [
           style = styleForToken;
         }
         if ((styleForToken !== style) &&
-            !((style === "setext_with_atx") &&
+            !((style === "setext_with_atx" || style === "setext_with_atx_closed") &&
               (/h[12]/.test(token.tag) && (styleForToken === "setext")) ||
-               (/h[^12]/.test(token.tag) && (styleForToken === "atx")))) {
+              (/h[^12]/.test(token.tag) &&
+                  (styleForToken === "atx" || styleForToken === "atx_closed")))
```

Comment:
This if statement is getting a bit hairy, but it looked like you kept all of the rule logic together, so I didn't want to break it out into its own function.